### PR TITLE
Add a reference task to generate REFERENCE.md

### PR DESCRIPTION
**Pull request by ekohl**
In puppet-extlib we can now set:

```yaml
Rakefile:
  puppet_strings_patterns: lib/puppet/functions/extlib/*.rb
```

### CODE REVIEW

**Comment by ekohl** [comment]
I left it out here, but I think we can now modify our Travis release job as follows:

```
before_deploy: "bundle exec rake reference"
```

That should fix part of https://github.com/voxpupuli/modulesync_config/issues/397 by ensuring we always have an up to date REFERENCE.md on the forge for released modules.

**Comment by ekohl** [comment]
Btw, https://github.com/puppetlabs/puppet-strings/pull/192 is similar, but doesn't allow overriding the pattern via the config.

File: moduleroot/Rakefile.erb [hunk 1]
```diff
@@ -5,7 +5,16 @@ require 'puppetlabs_spec_helper/rake_tasks'
...
   require 'voxpupuli/release/rake_tasks'
+rescue LoadError
+end
+
+begin
   require 'puppet-strings/tasks'
+
+  desc 'Generate REFERENCE.md'
+  task :reference do
+    Rake::Task['strings:generate'].invoke('<%= @configs['puppet_strings_patterns'] %>', nil, nil, nil, 'false', 'true')
```
Comment (ghoneycutt) [inline moduleroot/Rakefile.erb]: This should document what the fields are or link to the documentation as this is hard to grok.

Comment (ekohl) [reply moduleroot/Rakefile.erb]: You're right. I initially attempted to pass in arguments as a named hash but it looks like the rake invoke task doesn't support this so I took this short cut.

**Comment by ekohl** [comment]
Updated. This now requires the freshly released puppet-strings 2.2.0 which includes my patch with a `strings:generate:reference`. That makes our invocation a lot easier and self documenting.

**Comment by ekohl** [comment]
https://github.com/voxpupuli/puppet-extlib/pull/131 is how puppet-extlib would make use of this.

File: config_defaults.yml [hunk 2]
```diff
@@ -101,6 +101,7 @@ Rakefile:
   - 'disable_single_quote_string_with_variables'
   default_enabled_rake_targets:
   - 'release_checks'
+  puppet_strings_patterns: ''
```
Comment (alexjfisher) [inline config_defaults.yml]: Should we make this an array instead?

Comment (ekohl) [reply config_defaults.yml]: I like that idea. Will update

**Comment by logicminds** [comment]
This should be

```
Rakefile:
  puppet_strings_patterns: lib/puppet/functions/**/*.rb functions/**/*.pp
```

**Comment by ekohl** [comment]
Not sure I follow. Why is that?

**Comment by logicminds** [comment]
Ahh nevermind that is the custom config for the extlib.

**Pull request merged**